apps/speed.c

```diff
@@ -480,155 +408,78 @@ static const OPT_PAIR doit_choices[] = {
     {"rand", D_RAND}
 };
 
-static double results[ALGOR_NUM][OSSL_NELEM(lengths_list)];
+static double results[ALGOR_NUM][SIZE_NUM];
 
 #ifndef OPENSSL_NO_DSA
-# define R_DSA_512       0
-# define R_DSA_1024      1
-# define R_DSA_2048      2
-static const OPT_PAIR dsa_choices[] = {
-    {"dsa512", R_DSA_512},
-    {"dsa1024", R_DSA_1024},
-    {"dsa2048", R_DSA_2048}
+enum { R_DSA_512, R_DSA_1024, R_DSA_2048, DSA_NUM };
+static const char * const dsa_choices[DSA_NUM] = {
+    "dsa512", "dsa1024", "dsa2048"
 };
```

Comment:
I *strongly* dislike that we now must make sure these kinds of arrays are kept in sync with the corresponding enum.  For a few items like this, it's not so bad, but for larger arrays, it gets unnecessarily burdensome.  Please bring back the OPT_PAIR setup.

enums are fine, I'm not asking you to bring back the macros.

Reply:
Fixed